torch_geometric/nn/aggr/adaptive.py

```diff
@@ -0,0 +1,400 @@
+                and use_batch_norm is None and dropout is None
+            ), ('Cannot specify MLP arguments if a custom MLP is provided.')
+
+        if num_hidden_layers is not None:
+            assert num_hidden_layers >= 0, (
+                'Number of hidden layers must be non-negative.')
+
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+        self.hidden_dim = hidden_dim
+        self.max_num_nodes_in_dataset = max_num_nodes_in_dataset
+        self.num_hidden_layers = num_hidden_layers
+        self.use_batch_norm = use_batch_norm
+        self.dropout = dropout
+        self.custom_mlp = custom_mlp
+
+        if self.custom_mlp is None:
+            # Input layer
+            modules = [
+                nn.Linear(
+                    in_features=self.max_num_nodes_in_dataset *
+                    self.in_channels, out_features=self.hidden_dim),
+                nn.BatchNorm1d(self.hidden_dim),
+                nn.ReLU(),
+            ]
+
+            # Hidden layers
+            for _ in range(self.num_hidden_layers):
+                modules.append(
+                    nn.Linear(in_features=self.hidden_dim,
+                              out_features=self.hidden_dim))
+                modules.append(
+                    nn.BatchNorm1d(self.hidden_dim) if self.
+                    use_batch_norm else nn.Identity())
+                modules.append(nn.ReLU())
+
+            # Output layer
+            modules.append(
+                nn.Linear(in_features=self.hidden_dim,
+                          out_features=self.out_channels))
+            modules.append(
+                nn.Dropout(p=dropout) if self.dropout else nn.Identity())
+
+            self.mlp_agg = nn.Sequential(*modules)
+        else:
+            self.mlp_agg = self.custom_mlp
+
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        for module in self.mlp_agg.children():
+            if hasattr(module, 'reset_parameters'):
+                module.reset_parameters()
+
+    def forward(self, x: Tensor, index: Optional[Tensor] = None,
+                ptr: Optional[Tensor] = None, dim_size: Optional[int] = None,
+                dim: int = -2) -> Tensor:
+        x, _ = to_dense_batch(x, index, fill_value=0,
+                              max_num_nodes=self.max_num_nodes_in_dataset)
+        return self.mlp_agg(x.view(-1, x.shape[1] * x.shape[2]))
+
+    def __repr__(self) -> str:
+        has_custom_mlp = self.custom_mlp is not None
+        return (f'{self.__class__.__name__}({self.max_num_nodes_in_dataset}, '
+                f'{self.in_channels}, {self.out_channels}, '
+                f'{self.num_hidden_layers}, {self.hidden_dim}, '
+                f'{self.use_batch_norm}, {self.dropout}, '
+                f'Uses custom MLP: {has_custom_mlp})')
+
+
+# Set Transformer code as provided in
+# https://github.com/juho-lee/set_transformer
+
+
+class MAB(nn.Module):
```

Comment:
Do you think we can reuse [this](https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/nn/aggr/gmt.py).

Reply:
I was not sure if it would be possible without refactoring the GMT code. The computation for the MAB itself should be the same but there is some additional logic in the GMT version. I haven't tested this but I think it may be possible with the right arguments.